src/config.rs

```diff
@@ -0,0 +1,211 @@
+use std::str::FromStr;
+
+use strum::{EnumProperty, IntoEnumIterator};
+use strum_macros::{AsRefStr, Display, EnumIter, EnumProperty, EnumString};
+
+use crate::constants::DC_VERSION_STR;
+use crate::context::Context;
+use crate::dc_job::*;
+use crate::dc_stock::*;
+use crate::dc_tools::*;
+use crate::x::*;
+
+/// The available configuration keys.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Display, EnumString, AsRefStr, EnumIter, EnumProperty,
+)]
+#[strum(serialize_all = "snake_case")]
+pub enum Config {
+    Addr,
+    MailServer,
+    MailUser,
+    MailPw,
+    MailPort,
+    SendServer,
+    SendUser,
+    SendPw,
+    SendPort,
+    ServerFlags,
+    #[strum(props(default = "INBOX"))]
+    ImapFolder,
+    Displayname,
+    Selfstatus,
+    Selfavatar,
```

Comment:
maybe `SelfStatus`, `SelfAvatar`?

Reply:
they were `selfstatus` and `selfavatar` before, and I wanted to use the same rules to capitalize and turn into strings for all members

Reply:
i would not change it. using a simple capitalization rule makes sense imho - otherwise there might be easy mismatches and confusion between the "object" and the strings really used in the database and in the ui.